riff-raff/app/schedule/DeployJob.scala

```diff
@@ -56,8 +52,12 @@ object DeployJob extends Logging with LogAndSquashBehaviour {
         } else {
           Left(Error(s"Scheduled deployments disabled. Would have deployed $params"))
         }
+      case RunState.Failed =>
+        Left(defaultError(RunState.Failed).copy(scheduledDeployError = Some(SkippedDueToPreviousFailure(lastDeploy))))
+      case RunState.NotRunning =>
+        Left(defaultError(RunState.Failed).copy(scheduledDeployError = Some(SkippedDueToPreviousWaitingDeploy(lastDeploy))))
```

Comment:
This looks like you should be passing in `RunState.NotRunning` here? The `defaultError` method actually has access to `lastDeploy.state` so might be easier to get it directly as per `uuid`.

Reply:
Thanks for spotting this - you caught me copy and pasting 🙈 

Fortunately, this problem goes away anyway now that we've moved to an error hierarchy.